### ISSUE TITLE
build error: cannot find pyarrow  build dir

### ISSUE THREAD

**Issue by elibol** [ray-project/ray#1197]
The pyarrow build directory is found by searching for the substring `lib`. The path it checks is absolute. If `lib` exists in the path to the pyarrow build dir, then you may get this error:

```
+ PYARROW_BUILD_LIB_DIR=/home/elibol/dev/ray/src/thirdparty/arrow/python/build
+ echo 'copying pyarrow files from /home/elibol/dev/ray/src/thirdparty/arrow/python/build/pyarrow'
copying pyarrow files from /home/elibol/dev/ray/src/thirdparty/arrow/python/build/pyarrow
+ cp -r /home/elibol/dev/ray/src/thirdparty/arrow/python/build/pyarrow /home/elibol/dev/ray/src/thirdparty/../../python/ray/pyarrow_files/
cp: cannot stat '/home/elibol/dev/ray/src/thirdparty/arrow/python/build/pyarrow': No such file or directory
Traceback (most recent call last):
  File "setup.py", line 121, in <module>
    license="Apache 2.0")
  File "/home/elibol/anaconda3/envs/ray/lib/python3.6/distutils/core.py", line 148, in setup
    dist.run_commands()
  File "/home/elibol/anaconda3/envs/ray/lib/python3.6/distutils/dist.py", line 955, in run_commands
    self.run_command(cmd)
  File "/home/elibol/anaconda3/envs/ray/lib/python3.6/distutils/dist.py", line 974, in run_command
    cmd_obj.run()
  File "/home/elibol/anaconda3/envs/ray/lib/python3.6/site-packages/setuptools/command/install.py", line 67, in run
    self.do_egg_install()
  File "/home/elibol/anaconda3/envs/ray/lib/python3.6/site-packages/setuptools/command/install.py", line 109, in do_egg_install
    self.run_command('bdist_egg')
  File "/home/elibol/anaconda3/envs/ray/lib/python3.6/distutils/cmd.py", line 313, in run_command
    self.distribution.run_command(command)
  File "/home/elibol/anaconda3/envs/ray/lib/python3.6/distutils/dist.py", line 974, in run_command
    cmd_obj.run()
  File "/home/elibol/anaconda3/envs/ray/lib/python3.6/site-packages/setuptools/command/bdist_egg.py", line 169, in run
    cmd = self.call_command('install_lib', warn_dir=0)
  File "/home/elibol/anaconda3/envs/ray/lib/python3.6/site-packages/setuptools/command/bdist_egg.py", line 155, in call_command
    self.run_command(cmdname)
  File "/home/elibol/anaconda3/envs/ray/lib/python3.6/distutils/cmd.py", line 313, in run_command
    self.distribution.run_command(command)
  File "/home/elibol/anaconda3/envs/ray/lib/python3.6/distutils/dist.py", line 974, in run_command
    cmd_obj.run()
  File "/home/elibol/anaconda3/envs/ray/lib/python3.6/site-packages/setuptools/command/install_lib.py", line 11, in run
    self.build()
  File "/home/elibol/anaconda3/envs/ray/lib/python3.6/distutils/command/install_lib.py", line 107, in build
    self.run_command('build_ext')
  File "/home/elibol/anaconda3/envs/ray/lib/python3.6/distutils/cmd.py", line 313, in run_command
    self.distribution.run_command(command)
  File "/home/elibol/anaconda3/envs/ray/lib/python3.6/distutils/dist.py", line 974, in run_command
    cmd_obj.run()
  File "setup.py", line 50, in run
    subprocess.check_call(["../build.sh", sys.executable])
  File "/home/elibol/anaconda3/envs/ray/lib/python3.6/subprocess.py", line 291, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['../build.sh', '/home/elibol/anaconda3/envs/ray/bin/python']' returned non-zero exit status 1.
```

If you run into this issue, a temporary workaround is to replace line 99 in `src/thirdparty/build_thirdparty.sh` with:
```
TEMP_PWD=`pwd`
cd $TP_DIR/arrow/python/build
PYARROW_BUILD_LIB_DIR="$TP_DIR/arrow/python/build/$(find ./ -maxdepth 1 -type d -print | grep -m1 'lib')"
cd $TEMP_PWD
```

**Comment by robertnishihara** [comment]
Good catch! Grepping for `lib` was a hack that we should probably get rid of anyway.

**Comment by robertnishihara** [comment]
Fixed by #1354
